src/linux/InotifyTree.cpp

```diff
@@ -259,6 +339,11 @@ InotifyTree::InotifyNode::InotifyNode(
       emitCreatedEvent
     );
 
+    if (!mTree->addInode(file.st_ino, child)) { // Skip this inode if already watching
```

Comment:
If I'm understanding correctly, we're now adding the `child` to the `inodes` map. Down below on line 350, if the `child->isAlive()` check fails, we delete the child without removing it from the map, leaving a dangling pointer.

Reply:
Right, fixed in next commit